### DISCUSSION TITLE
installation guide for dummies?

### DISCUSSION THREAD

**Discussion by ioudcitysong** [krateng/maloja#371] [Q&A]
Look, I'm someone that loves lastfm, toying with its statistics and logging my listening habits -- of course, the site itself has its own limitations and flaws, so I'm always up looking for third-party sites & applications to enhance its experience. I stumbled onto Maloja and it seems great, I particularly love the featured artist functionality because it's something I know they'll never implement in lastfm.

Unfortunately, I know next to nothing about programming and this project seems more targeted at developers, which is ok, but I really wanted to give it a try. Looking at the "How to install" section really does nothing for me, as I have no idea where to even start. If the project's developer could do a more step-by-step hand-guiding guide in how to make this work and what I have to install/set-up in my computer I'd be very happy. Thanks anyway!

**Comment by duckcantcode** [comment]
I can help you at [my Discord server](https://github.com/krateng/maloja/issues/226#issuecomment-1657247504). You'll need Docker, preferably. On Linux or WSL

**Comment by mauriva** [comment]
Despite having some experience of using docker, I have not yet managed to install Maloja on an external server (an Ubuntu server on an Oracle free tier virtual machine to be precise), so I find myself requesting that the start page of this project include documentation for the step-by-step installation.
I have tried a docker-compose.yml file with this content:

`services:
  maloja:
    image: krateng/maloja:latest
    ports:
      - '42010:42010'
    volumes:
      - ./maloja/data:/data
    environment:
      - MALOJA_FORCE_PASSWORD=XXXXX # Set your admin password here
      - MALOJA_DATA_DIRECTORY=/data
`

The installation seems to go smoothly and the service appears to be active, but when I try to access the server from outside on the set port, I cannot reach the Maloja page.
I have opened port 42010, but I fear that something is missing to access the service from outside.
By the way, I don't like to specify the password in the file: isn't there a better and more secure way to set it?

**Comment by mauriva** [comment]
I add this.
In other project there is a line to set the IP address of the server.
For example, in the Multi-Scobbler docker-compose.yml file there is this section:

_environment:
            - BASE_URL=http://192.168.0.100:9078_

Here is possbile to set the IP address of the server.
I don't know if also for Maloja I need a similar setting.

**Comment by mauriva** [comment]
I found the problem of why my Maloja instance would not start (it was DNS-related).
I still find the installation instructions to be updated.
It would be useful to include an example of docker-compose.yml in the readme file.
In addition, it would also be good if the password could be set in another way: reporting the value ‘MALOJA_FORCE_PASSWORD=XXXXX’ in the compose file leaves this setting visible, e.g. by logging in with Portainer.
Is there not another, more secure way?

**Reply by krateng** [reply]
You can use docker secrets, or force the password to some temporary value on the first run, then remove the environment variable and change it from the web interface, then restart.
